utils/SnapPy/Snappy/MeteorologyCalculator.py

```diff
@@ -69,28 +69,40 @@ def __str__(self):
 class MeteorologyCalculator(abc.ABC):
     '''Base-class to pre-calculate/extract meteorology'''
     @staticmethod
-    def findGlobalData(res: GlobalMeteoResource, dtime: datetime):
-        '''Method to find the closest global dataset earlier than dtime.
+    def findAllGlobalData(res: GlobalMeteoResource): 
+        '''Static method to find the all global dataset
 
         Args:
-            res: resources from getGlobalMeteoResources
             dtime: datetime object with a start-time, which should be included in the dataset
+            count: number of file
 
         Returns:
-            A tuple with referencetime and filename
-
-        Raises:
-            MeteoDataNotAvailableException: no data for the dtime can be found
+            A list of tuples with [(forecast-time, file)]
         '''
-        #should eventually use productstatus.met.no?
-
         timesFiles = [] # tuples with datetime, file
         for inDir in res.indirs:
             for iFile in iglob(os.path.join(inDir, res.pathglob)):
                 statinfo = os.stat(iFile)
                 if statinfo.st_mtime < (time.time() - res.path_grace_period_sec): # file hasn't been changed in x sec
                     dateFile = datetime.strptime(os.path.basename(iFile), res.pathptime)
                     timesFiles.append((dateFile, iFile))
+        return timesFiles
+
+    @staticmethod
+    def findGlobalData(res: GlobalMeteoResource, dtime: datetime):
+        '''Method to find the closest global dataset earlier than dtime.
```

Comment:
`Method to find the global dataset with the latest forecast time which includes dtime'

Reply:
Much better.